--- a/tools/icu/icutrim.py	2020-11-03 16:54:23.000000000 +0100
+++ b/tools/icu/icutrim.py	2020-11-03 16:54:27.000000000 +0100
@@ -316,7 +316,10 @@
             erritems = fi.readlines()
             fi.close()
             #Item zone/zh_Hant_TW.res depends on missing item zone/zh_Hant.res
-            pat = re.compile(bytes(r"^Item ([^ ]+) depends on missing item ([^ ]+).*", 'utf-8'))
+            if str == bytes:
+              pat = re.compile(bytes(r"^Item ([^ ]+) depends on missing item ([^ ]+).*"))
+            else:
+              pat = re.compile(bytes(r"^Item ([^ ]+) depends on missing item ([^ ]+).*", 'utf-8'))
             for i in range(len(erritems)):
                 line = erritems[i].strip()
                 m = pat.match(line)
